time/src/test/java/net/jqwik/time/api/DurationTests.java

```diff
@@ -0,0 +1,594 @@
+
+				Duration start = Duration.ofSeconds(0, -101);
+				Duration end = Duration.ofSeconds(0, 100);
+
+				Arbitrary<Duration> durations = Times.durations().between(start, end);
+
+				assertAllGenerated(durations.generator(1000), random, durationGenerated -> {
+					assertThat(durationGenerated).isNotNull();
+					return true;
+				});
+
+			}
+
+			@Provide
+			Arbitrary<Duration> worstCase() {
+				return Times.durations().between(Duration.ZERO, Duration.ofSeconds(Long.MAX_VALUE - 2, 999_999_999));
+			}
+
+			@Provide
+			Arbitrary<Duration> worstCaseNegative() {
+				return Times.durations().between(DefaultDurationArbitrary.DEFAULT_MIN, Duration.ofSeconds(-2, 0));
+			}
+
+		}
+
+	}
+
+	@Group
+	class SimpleAnnotations {
+
+		@Property
+		void validDurationIsGenerated(@ForAll Duration duration) {
+			assertThat(duration).isNotNull();
+		}
+
+	}
+
+	@Group
+	class DurationMethods {
+
+		@Property
+		void between(@ForAll("durations") Duration start, @ForAll("durations") Duration end, @ForAll Random random) {
+
+			Assume.that(start.compareTo(end) <= 0);
+
+			Arbitrary<Duration> durations = Times.durations().between(start, end);
+
+			assertAllGenerated(durations.generator(1000), random, duration -> {
+				assertThat(duration.compareTo(start)).isGreaterThanOrEqualTo(0);
+				assertThat(duration.compareTo(end)).isLessThanOrEqualTo(0);
+			});
+
+		}
+
+		@Property
+		void betweenSame(@ForAll("durations") Duration durationSame, @ForAll Random random) {
+
+			Arbitrary<Duration> durations = Times.durations().between(durationSame, durationSame);
+
+			assertAllGenerated(durations.generator(1000), random, duration -> {
+				assertThat(duration).isEqualTo(durationSame);
+			});
+
+		}
+
+	}
+
+	@Group
+	class Shrinking {
+
+		@Property
+		void defaultShrinking(@ForAll Random random) {
+			DurationArbitrary durations = Times.durations();
+			Duration value = falsifyThenShrink(durations, random);
+			assertThat(value).isEqualTo(Duration.ofSeconds(0, 0));
+		}
+
+		@Property(tries = 40)
+		void shrinksToSmallestFailingPositiveValue(@ForAll Random random) {
+			DurationArbitrary durations = Times.durations();
+			TestingFalsifier<Duration> falsifier = duration -> duration.compareTo(Duration.ofSeconds(999_392_192, 709_938_291)) < 0;
+			Duration value = falsifyThenShrink(durations, random, falsifier);
+			assertThat(value).isLessThanOrEqualTo(Duration.ofSeconds(999_392_193));
+			assertThat(value).isGreaterThanOrEqualTo(Duration.ofSeconds(999_392_192, 709_938_291));
+		}
+
+		@Property(tries = 10)
+		void shrinksToSmallestFailingNegativeValue(@ForAll Random random) {
+			DurationArbitrary durations = Times.durations();
+			TestingFalsifier<Duration> falsifier = duration -> duration.compareTo(Duration.ofSeconds(-999_392_192, 709_938_291)) > 0;
+			Duration value = falsifyThenShrink(durations, random, falsifier);
+			assertThat(value).isEqualTo(Duration.ofSeconds(-999_392_192));
+		}
+
+	}
+
+	@Group
+	class ExhaustiveGeneration {
```

Comment:
Exhaustive generation is especially interesting for reduced precision, e.g. to minutes. Those are IMO the more interesting tests. Also numbered tests again.

Reply:
There isn't the method ofPrecision at the moment. Durations only have two methods: `getSeconds()` and `getNanos()` so I thought it is not necessary. Should I add it?

Reply:
I have the between test cases at ExhaustiveGeneration because I want to know if really all values can be generated.

Reply:
Yes, I think DurationArbitrary should get an ofPrecision()

Reply:
Okay I'll work on it. If you'd like, you can proceed with your review while I work on ofPrecision.